From 4e44d5c41ff9d97fdae4e0951ef1d461e07e84ad Mon Sep 17 00:00:00 2001
From: zhongjiawei <zhongjiawei1@huawei.com>
Date: Thu, 5 Jan 2023 16:25:57 +0800
Subject: [PATCH] runc: ignore exec.fifo removing not exist error

---
 libcontainer/container_linux.go | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/libcontainer/container_linux.go b/libcontainer/container_linux.go
index 9df830d..6b68a0e 100644
--- a/libcontainer/container_linux.go
+++ b/libcontainer/container_linux.go
@@ -325,7 +325,10 @@ func handleFifoResult(result openResult) error {
 	if err := readFromExecFifo(f); err != nil {
 		return err
 	}
-	return os.Remove(f.Name())
+	if err := os.Remove(f.Name()); !os.IsNotExist(err) {
+		return err
+	}
+	return nil
 }
 
 type openResult struct {
-- 
2.30.0

