From b8495e4a88faa18a3851235b9b2431164826cc3f Mon Sep 17 00:00:00 2001
From: zhong-jiawei-1 <zhongjiawei1@huawei.com>
Date: Fri, 21 Oct 2022 15:37:47 +0800
Subject: [PATCH] runc: modify minHookTimeOut

---
 runc-1.1.3/libcontainer/configs/config.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/runc-1.1.3/libcontainer/configs/config.go b/runc-1.1.3/libcontainer/configs/config.go
index 9076846..540bcdb 100644
--- a/runc-1.1.3/libcontainer/configs/config.go
+++ b/runc-1.1.3/libcontainer/configs/config.go
@@ -12,7 +12,7 @@ import (
 )
 
 const (
-	minHookTimeOut     = 1 * time.Second
+	minHookTimeOut     = 0
 	defaultHookTimeOut = 5 * time.Second
 	maxHookTimeOut     = 120 * time.Second
 	//the runc default timeout is 120s, so set the defaultWarnTime to 80% of the default timeout.
-- 
2.30.0

